### PR TITLE
Hosting a Webserver Using a Raspberry Pi

### PR DESCRIPTION
**Pull request by eunice-wanjiku**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
 
This solves #3008.      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
@eunice-wanjiku 
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Review by Dawe-7 — COMMENTED** [review]
Hello @eunice-wanjiku - Welcome to Section's engineering education program. I'll be reviewing your article. Before we begin the review, i need you to make the following changes:
1. Use subheadings format for listing the steps. Use `###` for headings and `####` or lower for subheadings.  The steps heading should give a brief description of what is happening in each step. Take a look at this [tutorial](https://github.com/section-engineering-education/engineering-education/pull/1250/files) to understand this better.
2. I also noticed some of your images are in the correct file format but are in capital. 
As soon as you make the changes we'll begin the review. Also revisit your article one more time to make the review process faster.

**Comment by eunice-wanjiku** [comment]
> Hello @eunice-wanjiku - Welcome to Section's engineering education program. I'll be reviewing your article. Before we begin the review, i need you to make the following changes:
> 
> 1. Use subheadings format for listing the steps. Use `###` for headings and `####` or lower for subheadings.  The steps heading should give a brief description of what is happening in each step. Take a look at this [tutorial](https://github.com/section-engineering-education/engineering-education/pull/1250/files) to understand this better.
> 2. I also noticed some of your images are in the correct file format but are in capital.
>    As soon as you make the changes we'll begin the review. Also revisit your article one more time to make the review process faster.

@Dawe-7 Done

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Dawe-7** [comment]
> > Hello @eunice-wanjiku - Welcome to Section's engineering education program. I'll be reviewing your article. Before we begin the review, i need you to make the following changes:
> > 
> > 1. Use subheadings format for listing the steps. Use `###` for headings and `####` or lower for subheadings.  The steps heading should give a brief description of what is happening in each step. Take a look at this [tutorial](https://github.com/section-engineering-education/engineering-education/pull/1250/files) to understand this better.
> > 2. I also noticed some of your images are in the correct file format but are in capital.
> >    As soon as you make the changes we'll begin the review. Also revisit your article one more time to make the review process faster.
> 
> @Dawe-7 Done

@eunice-wanjiku, Kindly revisit the suggestions once more because i can see some of the changes requested have not been implemented. Also, revisiting your article and making changes where necessary will make the review process faster. Consider using the [hemimgway editor](https://hemingwayapp.com/) and [grammarly](https://www.grammarly.com/) to polish up your article.

**Comment by eunice-wanjiku** [comment]
@Dawe-7 Kindly check the updated files. I have uploaded the revision as requested

File: content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/hosting-a-webserver-using-a-raspberry-pi/hero.jpg
+    alt: Hosting a Webserver Using a Raspberry Pi Hero Image
+---
+The Raspberry Pi can be used as a web server on your main local network or the internet at large. It is a great selection in cases where you want an Intranet for the office or web development server.
+<!--more-->
+You can create a local Pi webserver to deliver various contents while you are continuously surfing over the internet. To make the webserver operational, the Raspberry Pi should be connected to the local network and ensured that it has the latest version of the operating system of Raspbian. These are the defined instructions that you need for any model to work even on the pocket-sized Raspberry Pi Zero W and the very powerful Raspberry Pi 4.
```
Comment (Dawe-7) [inline content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md:19]:
```suggestion
You can create a local Pi webserver to deliver various contents while you are surfing over the internet. To make the webserver operational, the Raspberry Pi should be connected to the local network.  And, ensured that it has the latest version of the operating system of **Raspbian**. These are the defined instructions that you need for any model to work even on the pocket-sized Raspberry Pi Zero W and the very powerful Raspberry Pi 4.
```

File: content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+### Table of contents
+- [Prerequisites](#prerequisites)
+- [Introduction](#introduction)
+- [Setting up Apache on Raspberry Pi](#setting-up-apache-on-raspberry-pi)
+- [Changing The Host Name of the Server](#changing-the-host-name-of-the-server)
+- [Conclusion](#conclusion)
+- [Relevant resources](#relevant-resources)
+
+### Prerequisites
+For the reader to follow through and be guided accordingly, they are assumed to have an installed operating system with the Pixel window manager. In summary, you need to have;
```
Comment (Dawe-7) [inline content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md:30]:
```suggestion
For the reader to follow through, they are assumed to have an installed operating system with the Pixel window manager. In summary, you need to have:
```

File: content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+After PHP has been successfully installed, we can now download our desired database. In this case, this tutorial uses the MySQL database. Therefore, this step describes how to download MySQL.
+To do this, run the following command in the terminal; 
+
+````
+sudo apt-get install MySQL-server
+````
+
+![My SQL Download](/engineering-education/hosting-a-webserver-using-a-raspberry-pi/downloaddb.png)
+ 
+Note: When prompted to change the root password, please choose a strong password for MySQL and make sure to write it down somewhere you can refer to later on.
```
Comment (Dawe-7) [inline content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md]:
```suggestion
>Note: When prompted to change the root password, please choose a strong password for MySQL. Make sure to write it down somewhere you can refer to later on.
```

File: content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+2. You can also opt to connect remotely through SSH in case it is enabled on the Raspberry Pi. This is possible through a tool such as Putty and then use raspberry’s username and password to log in.
+
+Note: The default credentials for login into a raspberry Pi using SSH are as follows:
+Username: pi
+Password: raspberry
+
+![Putty Connection](/engineering-education/hosting-a-webserver-using-a-raspberry-pi/puttylogin.png)
+
+#### Step 2: Updating packages
+In this step, we will make sure that the available packages in our Raspberry pi are updated by typing the below command into the Terminal. This ensures you have the current versions of any file you download afterward.
```
Comment (Dawe-7) [inline content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md:55]:
```suggestion
In this step, we will make sure that the available packages in our Raspberry pi are updated by typing the command below into the Terminal. This ensures you have the current versions of any file you download afterwards.
```

File: content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+After PHP has been successfully installed, we can now download our desired database. In this case, this tutorial uses the MySQL database. Thus, this step describes how to download MySQL.
+To do this, run the following command in the terminal:
+
+```
+sudo apt-get install MySQL-server
+```
+
+![My SQL Download](/engineering-education/hosting-a-webserver-using-a-raspberry-pi/downloaddb.png)
+
+Note: When prompted to change the root password, please choose a strong password for MySQL and make sure to write it down somewhere you can refer to later on.
```
Comment (Dawe-7) [inline content/articles/hosting-a-webserver-using-a-raspberry-pi/index.md:92]:
```suggestion
>Note: When prompted to change the root password, please choose a strong password for MySQL. Make sure to write it down somewhere you can refer to later on.
```

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
@eunice-wanjiku, I've left you a few suggestions which will improve the quality of your article. Consider making the changes and we can move on to the next step. Also, identify your terminal commands output as `bash`. I have illustrated that in `step 2` so you understand and do the same for the rest of the commands. Adding some more content in your conclusion will also be helpful.

**Comment by eunice-wanjiku** [comment]
@Dawe-7 Adjusted the article as requested. Kindly verify

**Review by Dawe-7 — APPROVED** [review]
@eunice-wanjiku, Thanks for the changes, the article looks great.

@hectorkambow, this article is ready for final review.

**Comment by ahmadmardeni1** [comment]
Congrats on your first article and welcome to Section EngEd 🎉 @eunice-wanjiku
Please go through the changes I made @eunice-wanjiku @Dawe-7 

Thank you for the review @Dawe-7

**Pull request merged**